iothub/service/src/DirectMethod/Models/DirectMethodClientResponse.cs

```diff
@@ -28,13 +28,13 @@ protected internal DirectMethodClientResponse()
         public int Status { get; protected internal set; }
 
         /// <summary>
-        /// Get the payload object. May be null or empty.
+        /// Get the payload as a JSON string.
         /// </summary>
         /// <remarks>
-        /// The payload can be null or primitive type (e.g., string, int/array/list/dictionary/custom type)
+        /// To get the payload as a specified type, use <see cref="TryGetPayload{T}(out T)"/>.
         /// </remarks>
         [JsonIgnore]
-        public object Payload => JsonConvert.DeserializeObject((string)JsonPayload.Value);
+        public string PayloadAsString => JsonPayload.Value<string>();
```

Comment:
This doesn't work as well as we'd like, and it often exposes JArray and JObject types.

So, let's give them the serialized string and add a `TryGetValue<T>()` like we have done elsewhere.